app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -32,7 +32,7 @@ def destroy
 
     def render_success
       if skip_authorization? || can_authorize_response?
-        redirect_or_render(authorize_response)
```

Comment:
Can we better extract `redirect_or_render(authorize_response)` into a method like `respond_with_authorization` or similar to avoid action calling inside another methods/actions?

Reply:
Possibly, but `redirect_or_render` is also another `private` in the same controller, does not live anywhere else and calls controller-owned methods ... I don't have any good a ideas for where a public method would get defined? Calling `create` seems public enough behavior?

For our purposes, we created our own `AuthorizationsController` class that inherits from `Doorkeeper`, modified the routes accordingly and then overrode `create`:

```
module Ours
  class AuthorizationsController < Doorkeeper::AuthorizationsController
    # Override app/controllers/doorkeeper/authorizations_controller.rb

    # Override the #create method
    # To catch and render errors for the ActiveRecord validation
    def create
      super # <= This is why we're proposing this pull request
    rescue ActiveRecord::RecordInvalid => exception
      case exception.message
      when /Validation failed: Tenant has already been taken/
        pre_auth.error = Ours::Errors::DuplicateAuthorization
        redirect_or_render(pre_auth.error_response)
      else
        raise exception
      end
    end
  end
end
```

Reply:
@nbulaj Any more thoughts on this? We've been running this on a private branch for 3 weeks and haven't encountered any adverse side effects.

Reply:
Not sure @kmayer what is the problem with the private methods which is allowed to be overridden in Ruby . I still don't find it a good choice to call `#create` action inside another method in the controller (like any other actions)